### PR TITLE
vimc-5014 basic functionality to create YT tickets on successful report creation

### PR DESCRIPTION
**Pull request by hillalex**
This adds the most basic first step - ticket creation on success, with a link to the report and person assigned, and tags for disease/group/touchstone.

Further PRs to follow in this repo for:
* creating ticket on failure as well
* relating tickets of same group/diseaset/touchstone combo to each other

Also needs a Montagu PR for getting the YouTrack token into the config during deployment.

### CODE REVIEW

File: src/task_run_diagnostic_reports.py [hunk 1]
```diff
@@ -53,6 +58,21 @@ def success_callback(report, version):
         return {}
 
 
+def create_ticket(group, disease, touchstone,
+                  report: ReportConfig, version,
+                  yt: YTClient,
+                  config: Config):
+    try:
+        issue = yt.create_issue(Project("78-0"),
```
Comment (hillalex) [inline src/task_run_diagnostic_reports.py:66]: This is the "VIMC" project code - seems fine to hard-code as can't envision this changing

File: config/config.yml [hunk 2]
```diff
@@ -6,6 +6,8 @@ servers:
     password: password
   orderlyweb:
     url: http://localhost:8888
+  youtrack:
+    token: None
```
Comment (hillalex) [inline config/config.yml:10]: string value here for use in unit tests

File: test/integration/test_task_run_diagnostic_reports.py [hunk 3]
```diff
@@ -1,19 +1,37 @@
...
 
 @pytest.fixture(scope="module", autouse=True)
-def mod_header(request):
+def cleanup_emails():
     smtp.delete_all()
 
 
+@pytest.fixture(autouse=True)
+def cleanup_tickets():
+    issues = yt.get_issues("summary: {}".format(test_touchstone))
```
Comment (EmmaLRussell) [inline test/integration/test_task_run_diagnostic_reports.py:26]: Is this searching for issues to delete which have a summary containing the test touchstone? I thought it was the report version url which is being used for the summary?
Anyway, there do seem to be a lot of these tickets hanging around: https://mrc-ide.myjetbrains.com/youtrack/issues/VIMC?q=Check%20%26%20share

**Review by EmmaLRussell — COMMENTED** [review]
I knew you had this working because I got all the notification emails from YouTrack!

> Further PRs to follow in this repo for:
    creating ticket on failure as well
    relating tickets of same group/diseaset/touchstone combo to each other
Also needs a Montagu PR for getting the YouTrack token into the config during deployment.

Have we got tickets for all these? I guess the Montagu ticket (or a separate one) will also need to get all the real assignees into the config too?

This all looks good to me, I just got slightly confused over summary vs description when searching for the tickets to clean up - maybe a comment in the ticket creation line about which parameter is which would be good?

There seem to be a lot of old tickets hanging around - I think most of them are using an earlier version of the touchstone: https://mrc-ide.myjetbrains.com/youtrack/issues/VIMC?q=Check%20%26%20share

I wonder if it would be slightly safer to use a more exact test to find tickets to clean up to avoid accidentally deleting real tickets!

**Comment by hillalex** [comment]
> Have we got tickets for all these? I guess the Montagu ticket (or a separate one) will also need to get all the real assignees into the config too?

I was actually doing these under the same ticket, just different PRs. Could create new tickets though if that's clearer.

> I wonder if it would be slightly safer to use a more exact test to find tickets to clean up to avoid accidentally deleting real tickets! Oh I guess that was the point of using the rather specific touchstone version though?

Yeah, that's why I used this touchstone name. We could use the touchstone tag rather than summary to look them up, I think it was just throwing an error before the tag existed but I guess it exists now!

**Comment by hillalex** [comment]
And yeah looks like some other tests are creating tickets without a cleanup step, will have a look

**Comment by EmmaLRussell** [comment]
> I was actually doing these under the same ticket, just different PRs. Could create new tickets though if that's clearer.
> 
Don't really mind - there was a bit of discussion of this in recent months, and I think the consensus was to have separate tickets in general and keep a 1-1 relationship between tickets and PRs, to avoid confusion of what state a ticket should be in, and allow the webhook to reflect ticket state accurately.

**Review by EmmaLRussell — APPROVED** [review]

**Pull request merged**